gcd/src/main/java/org/spine3/server/storage/datastore/DsEntityStorage.java

```diff
@@ -60,43 +62,43 @@ private DsEntityStorage(Descriptor descriptor, DatastoreWrapper datastore) {
         this.datastore = datastore;
     }
 
+    @Nullable
     @Override
-    public M read(I id) {
-
-        final String idString = idToString(id);
+    protected EntityStorageRecord readInternal(I i) {
+        final String idString = idToString(i);
         final Key.Builder key = createKey(idString);
         final LookupRequest request = LookupRequest.newBuilder().addKey(key).build();
 
         final LookupResponse response = datastore.lookup(request);
 
         if (response == null || response.getFoundCount() == 0) {
-            return getEmptyMessage();
+            return getEmptyRecord();
         }
+
         final EntityResult entity = response.getFound(0);
-        final M result = entityToMessage(entity, typeName.toTypeUrl());
+
+        final EntityStorageRecord result = entityToMessage(entity, entityStorageRecordTypeName.toTypeUrl());
         return result;
     }
 
     @Override
-    public void write(I id, M message) {
-
-        checkNotNull(id, "Id is null.");
-        checkNotNull(message, "Message is null.");
+    protected void writeInternal(I i, EntityStorageRecord entityStorageRecord) {
```

Comment:
Please rename to `id` as we have conventions, the scope is too big for such a short name, and `i` is reserved name for loop iterators.